scaffolding-inspec/lib/scaffolding.sh

```diff
@@ -0,0 +1,209 @@
+  if [ "$(grep "compliance: " "$PLAN_CONTEXT/../inspec.yml")" ]; then
+    _do_compliance_login;
+  fi
+}
+
+do_default_setup_environment() {
+  set_buildtime_env PROFILE_CACHE_DIR "$HAB_CACHE_SRC_PATH/$pkg_dirname"
+  set_buildtime_env ARCHIVE_NAME "$pkg_name-$pkg_version.tar.gz"
+
+  # InSpec loads `pry` which tries to expand `~`. This fails if HOME isn't set.
+  set_runtime_env HOME "$pkg_svc_var_path"
+
+  # InSpec will create a `.inspec` directory in the user's home directory.
+  # This overrides that to write to a place within the running service's path.
+  # NOTE: Setting HOME does the same currently. This is here to be explicit.
+  set_runtime_env INSPEC_CONFIG_DIR  "$pkg_svc_var_path"
+}
+
+do_default_unpack() {
+  # Change directory to where the profile files are
+  pushd "$PLAN_CONTEXT/../" > /dev/null
+
+  # Get a list of all files in the profile except those that are Habitat related
+  profile_files=($(ls -I habitat -I results -I "*.hart"))
+
+  mkdir -p "$HAB_CACHE_SRC_PATH/$pkg_dirname" > /dev/null
+
+  # Copy just the profile files to the profile cache directory
+  cp -r ${profile_files[@]} "$HAB_CACHE_SRC_PATH/$pkg_dirname"
+}
+
+do_default_build_service() {
+  ## Create Hooks
+  mkdir -p "$pkg_prefix/hooks"
+  chmod 0750 "$pkg_prefix/hooks"
+
+  # Run hook
+  cat << EOF >> "$pkg_prefix/hooks/run"
+#!/bin/sh
+
+export HOME="{{pkg.svc_var_path}}"
+export INSPEC_CONFIG_DIR="{{pkg.svc_var_path}}"
+
+CFG_SPLAY_FIRST_RUN={{cfg.splay_first_run}}
+CFG_SPLAY_FIRST_RUN="\${CFG_SPLAY_FIRST_RUN:-0}"
+CFG_INTERVAL={{cfg.interval}}
+CFG_INTERVAL="\${CFG_INTERVAL:-300}"
+CFG_SPLAY={{cfg.splay}}
+CFG_SPLAY="\${CFG_SPLAY:-300}"
+CFG_LOG_LEVEL={{cfg.log_level}}
+CFG_LOG_LEVEL="\${CFG_LOG_LEVEL:-warn}"
+CFG_CHEF_LICENSE={{cfg.chef_license.acceptance}}
+CFG_CHEF_LICENSE="\${CFG_CHEF_LICENSE:-undefined}"
+CONFIG="{{pkg.svc_config_path}}/inspec_exec_config.json"
+PROFILE_PATH="{{pkg.path}}/{{pkg.name}}-{{pkg.version}}.tar.gz"
+
+inspec_cmd()
+{
+  inspec exec \${PROFILE_PATH} --json-config \${CONFIG} --chef-license \$CFG_CHEF_LICENSE --log-level \$CFG_LOG_LEVEL
+}
+
+
+SPLAY_DURATION=\$(shuf -i 0-\$CFG_SPLAY -n 1)
+SPLAY_FIRST_RUN_DURATION=\$(shuf -i 0-\$CFG_SPLAY_FIRST_RUN -n 1)
+
+exec 2>&1
+sleep \$SPLAY_FIRST_RUN_DURATION
+inspec_cmd
+
+while true; do
+  SLEEP_TIME=\$((\$SPLAY_DURATION + \$CFG_INTERVAL))
+  echo "InSpec is sleeping for \$SLEEP_TIME seconds"
+  sleep \$SPLAY_DURATION
+  sleep \$CFG_INTERVAL
+  inspec_cmd
+done
+EOF
+  chmod 0750 "$pkg_prefix/hooks/run"
+}
+
+do_default_build() {
+  inspec archive "$HAB_CACHE_SRC_PATH/$pkg_dirname" \
+                 --overwrite \
+                 -o "$HAB_CACHE_SRC_PATH/$pkg_dirname/$pkg_name-$pkg_version.tar.gz" \
+                 --chef-license "$CHEF_LICENSE"
+}
+
+do_default_install() {
+  cp "$HAB_CACHE_SRC_PATH/$pkg_dirname/$pkg_name-$pkg_version.tar.gz" "$pkg_prefix"
+
+  mkdir -p "$pkg_prefix/config"
+  chmod 0750 "$pkg_prefix/config"
+  cat << EOF >> "$pkg_prefix/config/cli_only.json"
```

Comment:
No rush - and certainly no blocker on this PR - but I'd really like to collaborate on how we can safely introduce changes to the JSON interfaces - the config InSpec accepts, and the reports it emits. Here, I see you're using the legacy format  (which is fine). 
We actually accept a more recent format of the config, which is mainly extended for handling complex scanning credentials like those needed for *scanning* a habitat installation). We added a version number to the format, so we can identify what version people are asking us to deal with, so we can validate it. We'll support the legacy format for a long time (maybe forever!), but eventually there may be features that you need in a future config format - and I think versioned config formats are one safe way to introduce change.